codejail/proxy.py

```diff
@@ -111,23 +111,20 @@ def get_proxy():
 
     # If we need a proxy, make a proxy.
     if PROXY_PROCESS is None:
-        # Start the proxy by invoking proxy_main.py in our root directory.
-        root = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
-        proxy_main_py = os.path.join(root, "proxy_main.py")
-
         # Run proxy_main.py with the same Python that is running us. "-u" makes
         # the stdin and stdout unbuffered. We pass the log level of the
         # "codejail" log so that the proxy can send back an appropriate level
         # of detail in the log messages.
         log_level = log.getEffectiveLevel()
-        cmd = [sys.executable, '-u', proxy_main_py, str(log_level)]
+        cmd = [sys.executable, '-u', '-c',
+               "from codejail import proxy; import sys; sys.exit(proxy.proxy_main(sys.argv))", str(log_level)]
```

Comment:
Another-- perhaps more idiomatic --way to do this would be to bring back proxy_main.py and move it to codejail/proxy_main.py, and there here to have:
```python
cmd = [sys.executable, '-u', '-m', 'codejail.proxy_main', str(log_level)]
```
The `-m` here tells Python to run a module.

I think it is functionally equivalent to what you have here. Up to you!